src/runner/runESLint.js

```diff
@@ -96,8 +98,55 @@ const getComputedFixValue = ({ fix, quiet, fixDryRun }) => {
   return undefined;
 };
 
+const useFlatConfig = async () =>
+  !!(await findUp('eslint.config.js', { cwd: process.cwd() }));
+
+const getESLintConstructor = async () => {
+  if (await useFlatConfig()) {
+    // Use a dynamic require here rather than a global require because this
+    // import path does not exist in eslint v7 which this library still
+    // supports
+    //
+    // Eslint exposes the new FlatESLint API under `eslint/use-at-your-own-risk` by
+    // using it's [export configuration](https://tinyurl.com/2s45zh9b).  However,
+    // the `import/no-unresolved` rule is [not aware of
+    // `exports`](https://tinyurl.com/469djpx3) and causes a false error here.  So,
+    // let's ignore that rule for this import.
+    //
+    // eslint-disable-next-line global-require, import/no-unresolved
+    return require('eslint/use-at-your-own-risk').FlatESLint;
+  }
+
+  return ESLint;
+};
+
+// Remove options that are not constructor args.
+const getESLintConstructorArgs = async cliOptions => {
+  // these are not constructor args for either the legacy or the flat ESLint
+  // api
+  const { fixDryRun, format, maxWarnings, quiet, ...legacyConstructorArgs } =
+    cliOptions;
```

Comment:
```suggestion
  const {
    fixDryRun,
    format,
    maxWarnings,
    quiet,
    ...legacyConstructorArgs
  } = cliOptions;
```

Reply:
hmm I think this one was done by prettier automatically....

Reply:
yeah prettier often makes code uglier, but i hope it wouldn't reformat my suggestion?

Reply:
I actually had it as you have it originally and this is what prettier reformatted it to

Reply:
gross, ok, guess we'll have to live with the uglier code :-/